src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -227,32 +245,62 @@ export const Map: React.FC<PropsType> = ({
       )),
     [attendances, setIsRoomModalOpen, setSelectedRoom, venue]
   );
+
+  const zoomIn = useCallback(() => {
+    setZoom((zoom) => Math.min(zoom + ZOOM_INCREMENT, MAX_ZOOM));
+  }, []);
+
+  const zoomOut = useCallback(() => {
+    setZoom((zoom) => Math.max(zoom - ZOOM_INCREMENT, MIN_ZOOM));
+  }, []);
+
   if (!user || !venue) {
     return <>Loading map...</>;
   }
 
+  const translateX =
+    (((mapSize.width * zoom - (mapSize.width * zoom) / zoom) / zoom) * 2) / 4;
+  const translateY =
+    (((mapSize.height * zoom - (mapSize.height * zoom) / zoom) / zoom) * 2) / 4;
```

Comment:
There's a lot of maths going on here.. can we simplify it at all? If not (or as well as), we should add some comments to help explain whats going on here.